### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by shegy28**
### Blog App - processing data in models
**In this repo I implemented the following:**

- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Create at least one user
- Create at least 4 posts written by one of the users
- Create at least 6 posts comments for one of the posts 
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

Add custom methods

Users

- A method that returns the 3 most recent posts for a given user.

Posts

- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.

Comments

- A method that updates the comments counter for a post.

Likes

- A method that updates the likes counter for a post.
- Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments
```
Comment (BregornOriginal) [inline app/models/user.rb]: - Kindly add `foreign_key: 'author_id'`

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  has_many :likes
```
Comment (BregornOriginal) [inline app/models/user.rb]: - Kindly add `foreign_key: 'author_id'`

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (BregornOriginal) [inline app/models/comment.rb]: This relationship should be something like this one 

`belongs_to :author, class_name: 'User'`

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (BregornOriginal) [inline app/models/like.rb]: This relationship should be something like this one

`belongs_to :author, class_name: 'User'`

**Review by BregornOriginal — APPROVED** [review]
## Hi @shegy28  

## Good job so far!

## There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✔️ Linters are passing
✔️ Good commit history
✔️ README looks professional

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 2 above)
Comment (shegy28) [reply app/models/user.rb]: add it where exactly

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @shegy28, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well working on the changes requested by the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**